modules/signals/src/with-computed.ts

```diff
@@ -1,22 +1,49 @@
+import { computed, isSignal, Signal } from '@angular/core';
 import {
-  SignalsDictionary,
   SignalStoreFeature,
   SignalStoreFeatureResult,
   StateSignals,
 } from './signal-store-models';
 import { Prettify } from './ts-helpers';
 import { withProps } from './with-props';
 
+type ComputedResult<
+  ComputedSignals extends Record<
+    string | symbol,
+    Signal<unknown> | (() => unknown)
+  >
+> = {
+  [P in keyof ComputedSignals]: ComputedSignals[P] extends () => infer V
+    ? Signal<V>
+    : ComputedSignals[P];
```

Comment:
Provided `Signal`s are also compatible with `() => infer V` type, but we should not change their type (e.g. when `DeepSignal` is provided to `withComputed` by using `deepComputed` helper).

```suggestion
  [K in keyof ComputedInput]: ComputedInput[K] extends Signal<any>
    ? ComputedInput[K]
    : ComputedInput[K] extends () => infer V ? Signal<V> : never;
```

When adding a type test, please also add one computed signal by using `deepComputed` and one writable signal to ensure their type hasn't been changed.